spec/index.md

```diff
@@ -709,23 +717,61 @@ In order to make a query call to canister, the user makes a POST request to `/ap
 
 -   `arg` (`blob`): Argument to pass to the canister method.
 
-If the call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
+Canister methods that do not change the canister state (except for cycle balance change due to message execution) can be executed more efficiently. This method provides that ability, and returns the canister's response directly within the HTTP response.
+
+If the query call resulted in a reply, the response is a CBOR (see [CBOR](#cbor)) map with the following fields:
 
--   `status` (`text`): `replied`
+-   `status` (`text`): `"replied"`
 
 -   `reply`: a CBOR map with the field `arg` (`blob`) which contains the reply data.
 
+-   (optional) `signatures` (`[+ node-signature]`): a singleton list of node signatures for the returned query response.
+
 If the call resulted in a reject, the response is a CBOR map with the following fields:
 
--   `status` (`text`): `rejected`
+-   `status` (`text`): `"rejected"`
 
 -   `reject_code` (`nat`): The reject code (see [Reject codes](#reject-codes)).
 
 -   `reject_message` (`text`): a textual diagnostic message.
 
 -   `error_code` (`text`): an optional implementation-specific textual error code (see [Error codes](#error-codes)).
 
-Canister methods that do not change the canister state (except for cycle balance change due to message execution) can be executed more efficiently. This method provides that ability, and returns the canister's response directly within the HTTP response.
+-   (optional) `signatures` (`[+ node-signature]`): a singleton list of node signatures for the returned query response.
```

Comment:
IMHO the formulation "singleton list" is somewhat ambiguous as it is unclear whether it refers to the list or its contents. IIUC we want it to refer to its contents, so maybe we can make this more clear.

Reply:
I've rephrased the formulation and also added a note why we have a list at the first place.